### PR TITLE
Profile details

### PR DESCRIPTION
**Pull request by yomnaafifi**
profile details apis for both customer and driver

### CODE REVIEW

File: customer/urls.py [hunk 1]
```diff
@@ -3,4 +3,8 @@
 
 urlpatterns = [
     path("signup/", views.CustomerSignupView.as_view(), name="customer_signup"),
+    path(
```
Comment (IxBahy) [inline customer/urls.py]: here the routes should be in theis format 
[
"customers/", views.ListAllCustomersView.as_view(), name="customer_details"
"customers/<int:id>", views.CustomerDetailsView.as_view(), name="customer_details"
]

2- the ListAllCustomersView should be called CustomersView  to have a more consistent naming

Comment (IxBahy) [reply customer/urls.py]: if this urls file have the "customers" prefix in the waddi.urls  already make it a  
[
"/", views.ListAllCustomersView.as_view(), name="customer_details"
"/int:id", views.CustomerDetailsView.as_view(), name="customer_details"
]

File: driver/urls.py [hunk 2]
```diff
@@ -2,4 +2,7 @@
 from . import views
 
 
-urlpatterns = [path("signup/", views.DriverSignupView.as_view(), name="driver_signup")]
+urlpatterns = [
+    path("signup/", views.DriverSignupView.as_view(), name="driver_signup"),
+    path("details/<int:id>", views.DriverDetailsView.as_view(), name="driver_details"),
```
Comment (IxBahy) [inline driver/urls.py]: same here remove the details if the urls have the "driver" prefix

File: customer/views.py [hunk 3]
```diff
@@ -43,3 +45,21 @@ def post(self, request, *args, **kwargs):
...
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ListAllCustomersView(generics.ListAPIView):
+    # this is only for testing purposes
+    queryset = CustomUser.objects.all()
+    serializer_class = ListCustomersSerializer
+
+
+class CustomerDetailsView(generics.RetrieveAPIView):
```
Comment (IxBahy) [inline customer/views.py]: why are u still using the  "generics.RetrieveAPIView" 
when this is based on the request.user and u anyway overrides the get method 
use generic api view insted for better flow

Comment (IxBahy) [reply customer/views.py]: and remove the id lookup

File: driver/views.py [hunk 4]
```diff
@@ -33,14 +35,26 @@ def post(self, request, *args, **kwargs):
...
                 driver = Driver.objects.create(user=user)
-                driver.preferred_method = request.data["preferred_method"]
+                driver.license = request.data["license"]
                 driver.save()
                 return Response(serializer.data, status=status.HTTP_201_CREATED)
         else:
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class DriverDetailsView(generics.RetrieveAPIView):
```
Comment (IxBahy) [inline driver/views.py]: why are u still using the  "generics.RetrieveAPIView" 
when this is based on the request.user and u anyway overrides the get method 
use generic api view insted for better flow 
and remove the id lookup

File: customer/urls.py (hunk 1 above)
Comment (yomnaafifi) [reply customer/urls.py]: yes kont lesa ha'ol

File: driver/views.py [hunk 5]
```diff
@@ -48,11 +46,10 @@ def post(self, request, *args, **kwargs):
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
-class DriverDetailsView(generics.RetrieveAPIView):
+class DriverDetailsView(generics.GenericAPIView):
```
Comment (IxBahy) [inline driver/views.py]: redundunt code rewrite in the auth model with a "profile/" route  and use it for both customer and driver

**Review by IxBahy — APPROVED** [review]

**Pull request merged**